File: 129-sum-root-to-leaf-numbers/129-sum-root-to-leaf-numbers.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sumNumbers(root *TreeNode) int {
    return sumSubtree(root, 0)
}



func sumSubtree(node *TreeNode, acc int) int {
    
    sum:=acc*10 + node.Val
    
    if node.Left == nil && node.Right == nil {
        return sum 
    }
    
    res:=0
    
    if node.Left != nil {
        res += sumSubtree(node.Left,  sum) 
    }
    
    if node.Right != nil {
        res += sumSubtree(node.Right, sum) 
    }
    
    return res 
}

File: 129-sum-root-to-leaf-numbers/129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        return self.sumSubtree(root, 0)
        
    def sumSubtree(self, node: Optional[TreeNode], acc: int) -> int:
        
        if node == None:
            return 0
        
        sum = acc*10 + node.val
        
        if (node.left == None) and (node.right == None):
            return sum
        
        return self.sumSubtree(node.left, sum) + self.sumSubtree(node.right, sum)